sdk/python/feast/infra/registry/base_registry.py

```diff
@@ -662,10 +663,16 @@ def to_dict(self, project: str) -> Dict[str, List[Any]]:
             key=lambda on_demand_feature_view: on_demand_feature_view.name,
         ):
             odfv_dict = self._message_to_sorted_dict(on_demand_feature_view.to_proto())
-
-            odfv_dict["spec"]["userDefinedFunction"][
+            # We are logging a warning because the registry object may be read from a proto that is not updated
```

Comment:
This `to_dict` method really annoys me, always complicates proto changes 😄 Do we actually need it anywhere? Let me look through the project to see if it's being used. Why can't we just convert the whole proto message to json and be done with it?

Reply:
For services that are python backed, they will likely use it. We do at Affirm.